lec17-rmarkdown.Rmd

```diff
@@ -1,4 +1,265 @@
+- For more recommendations on reproducible research, especially as it pertains to coding practices, read more [here](http://bioconnector.org/workshops/r-rmarkdown.html#some_recommendations_for_reproducible_research)
+
+### What are the barriers to reproducibility?
+
+- Brainstorm reasons why scientists are not readily embracing reproducible science.
+
+```{r, echo=FALSE}
+#Some ideas:
+#    - workflows are hard to document at every step
+#    - we already have systems in place that work, and would be hard/time consuming to upend them
+#    - people and skills, currently low adoption and hard to force collaborators to change
+```
+
+### Further Reading
+- This book called [Practice Reproducible Research](https://www.practicereproducibleresearch.org/)
+has many high level examples of how reproducible research looks in practice and further reading on the topic
+- Nature has a series of articles, editorials and research on the [challenges in  reproducible research](http://www.nature.com/news/reproducibility-1.17552)
+
+## R Markdown & Knitr
+
+R Markdown makes use of Pandoc's markdown formatting. We've seen a lot of the basic components to format our text so far, but to see the complete list, please visit the [official documentation](http://pandoc.org/MANUAL.html#pandocs-markdown).
+
+Before we start, everybody can do this 10 minute tutorial on markdown: http://commonmark.org/help/tutorial/
+
+### Lists, Tables, In-Line Code
+#### Lists
+**Unordered Lists (i.e. bullets)**
+```
+- Unordered list item
+    - Unordered list sub-item
+- Unordered list item
+```
+- Unordered list item
+    - Unordered list sub-item
+- Unordered list item
+
+**Ordered Lists (i.e. numbers)**
+```
+1. Ordered list item
+    1. Ordered list sub-item
+    2. Ordered list sub-item
+2. Ordered list item
+```
+1. Ordered list item
+    1. Ordered list sub-item
+    2. Ordered list sub-item
+2. Ordered list item
+
+#### Tables
+The `knitr` package has a function called `kable` that helps to display tables from an `r` code chunk nicely. It is best to use `echo=FALSE` and `results='asis'`.
+
+`r ''````{r, echo=FALSE, results='asis'}
+library(knitr)
+kable(head(mtcars))
+```
+
+You can also set the default data frame printing via the `df_print` option in your YAML metadata under `output` to do this automatically.
+
+```
+---
+title: Document
+output:
+    html_document:
+        df_print:kable
+---
+```
+
+#### In-line code
+
+If you want to state a value in your data in your text, it is best to reference to the actual variable or code containing that value rather than manually writing it out. Here is an example:
+
+There are `r ''``r nrow(df)` samples in this experiment.
+
+
+### Images and Figures
+To include an image, use the following syntax:
+```
+[caption for my image](path/to/image.png)
+```
+If you would like the caption you wrote to be included underneath your image, put the following in your YAML metadata:
+
+`fig_caption: yes`
```

Comment:
Maybe show the full yaml header, since fig_caption will need to be listed under the output document type